tests/management/workspace/test_model.py

```diff
@@ -37,90 +37,124 @@ def tearDown(self):
         error_messages = assertion.exception.messages
         self.assertEqual(len(error_messages), 1)
         self.assertIn("unique_default_root_workspace_per_tenant", error_messages[0])
 
     def test_single_default_per_tenant(self):
         """Test tenant can only have one default workspace"""
         with self.assertRaises(ValidationError) as assertion:
-            Workspace.objects.create(name="T1 Default Workspace Number 2", type="default", tenant=self.tenant)
+            Workspace.objects.create(
+                name="T1 Default Workspace Number 2",
+                type=Workspace.Types.DEFAULT,
+                tenant=self.tenant,
+                parent=self.tenant_1_root_workspace,
+            )
         error_messages = assertion.exception.messages
         self.assertEqual(len(error_messages), 1)
         self.assertIn("unique_default_root_workspace_per_tenant", error_messages[0])
 
-    def test_multiple_root_multiple_tenants(self):
-        """Test that multiple tenants can have more than one root workspace"""
-        try:
-            Workspace.objects.create(name="T2 Root Workspace Number 2", type="root", tenant=self.tenant_2)
-        except ValidationError as e:
-            self.fail("test_multiple_root_multiple_tenants raised ValidationError unexpectedly")
-
-    def test_multiple_default_multiple_tenants(self):
-        """Test that multiple tenants can have more than one default workspace"""
+    def test_multiple_root_and_default_multiple_tenants(self):
+        """Test that multiple tenants can have more than one root and default workspace"""
         try:
-            Workspace.objects.create(name="T2 Default Workspace Number 2", type="default", tenant=self.tenant_2)
+            tenant_2 = Tenant.objects.create(tenant_name="Tenant 2")
+            root = Workspace.objects.create(name="Root Workspace Number 2", type=Workspace.Types.ROOT, tenant=tenant_2)
+            default = Workspace.objects.create(
+                name="Default Workspace Number 2",
+                type=Workspace.Types.DEFAULT,
+                tenant=tenant_2,
+                parent=root,
+            )
         except ValidationError as e:
-            self.fail("test_multiple_default_multiple_tenants raised ValidationError unexpectedly")
+            self.fail("test_multiple_root_and_default_multiple_tenants raised ValidationError unexpectedly")
 
     def test_multiple_standard_per_tenant(self):
         """Test tenant can have multiple standard workspaces"""
         try:
             for n in ["1", "2", "3"]:
-                Workspace.objects.create(name=f"T1 Standard Workspace Number {n}", type="standard", tenant=self.tenant)
+                Workspace.objects.create(
+                    name=f"T1 Standard Workspace Number {n}",
+                    type=Workspace.Types.STANDARD,
+                    tenant=self.tenant,
+                    parent=self.tenant_1_default_workspace,
```

Comment:
Standard workspaces should be allowed to go underneath the root workspace, too. FWIW, I don't think I see any tests that exercise that. Not strictly related to this PR but might be good to add something for that.

Reply:
That's fair. I'll add a test case.